libass/ass_directwrite.c

```diff
@@ -677,7 +677,7 @@ static void match_fonts(void *priv, ASS_Library *lib,
 {
     ProviderPrivate *provider_priv = (ProviderPrivate *)priv;
     IDWriteFont *font = NULL;
-    IDWriteFontFamily *fontFamily = NULL;
+    IDWriteFontFamily *fontfamily = NULL;
```

Comment:
I think that naming style should be consistent throughout libass, i.e. always `snake_case` when possible (perhaps excluding copy-pasted prototypes, but I would have adapted even them). So I think that argument of `add_font` should be `font_family` (or maybe even simply `family`). It's not a target for this PR to fix style of old code, but I think all new variables should follow libass style conventions.

Reply:
Well, I always believe _local_ consistency trumps _global_ consistency. If this hunk is to introduce `font_family`, we’ll end up with two different names for the same thing in this file.

Reply:
> Well, I always believe _local_ consistency trumps _global_ consistency.

In that case style violation once introduced risks to never get dedicated cosmetic PR to fix it. Also such restyle-only commits complicate code history and blaming. I think incremental fixing around new changes is superior. And in this case I believe there are no consistency at all, as even local variables in the same function have named differently.

Reply:
This uses camel case for DirectWrite function arguments to match their declared formal parameter names, and it propagates these camel-case names throughout the code to ensure each name has exactly one spelling throughout the file. It uses snake case for other variables and fields to match libass’s default style.

This may have been a poor choice, but at least it makes some sense to me and is easy to explain and describe. If we start half-changing existing names, it will just be chaos instead.

For `fontFamily` especially, just searching for `fontFamily` within the file currently immediately brings up the full chain of control it goes through.

> Also such restyle-only commits complicate code history and blaming.

When a logic change touches a line anyway, then fixing minor style nits at the same time reduces noise. But with larger changes, I find it to be the exact opposite: when blaming, if I can tell that the last commit to change the line is purely stylistic from the commit message, I can immediately skip over it; but if it’s a commit that mixes logic with style, I must spend time figuring out why that commit touched this line.

In this particular case, I see a change to `fontFamily` as increasing entropy, not decreasing it. It would be interesting to hear the opinions of @rcombs and @TheOneric, as well as a word from @Apache553 who used neither option and wrote it as a single word :-)

Reply:
Since there seems to be some regularity to the current naming and to preserve the ability to find all instances of a variable being used *(even in other functions it's passed to)*, sticking to `fontFamily` in this commit seems reasonable to me.
Also adjusting all other instances of `fontFamily` in this commit would just introduce noise. If we want to change this, it should be done for the entire file and preferably in a separate commit.